internal/impl/io/input_socket_syslog.go

```diff
@@ -0,0 +1,252 @@
+
+	"github.com/jeromer/syslogparser/rfc3164"
+	"github.com/jeromer/syslogparser/rfc5424"
+)
+
+func init() {
+	err := service.RegisterInput(
+		"socket_syslog",
+		service.NewConfigSpec().
+			Summary("Creates a unix domain socket that receives syslog messages over a streaming or datagram socket.").
+			Description(`
+
+This input currently supports the `+"`rfc3164`"+` and `+"`rfc5424`"+` syslog header formats.
+
+If parsing fails, Benthos will raise an error. If you find this happening often, we recommend using the [socket_server](/docs/components/inputs/socket_server) input component combined with regex codec to implement your own parsing scheme.
+
+`).
+			Field(service.NewStringField("address").
+				Description("The address to listen from.").
+				Default("")).
+			Field(service.NewStringEnumField("type", "unix", "unixgram").
+				Description("Type of unix domain socket").
+				Default("unixgram")).
+			Field(service.NewIntField("max_buffer").
+				Description("The maximum message buffer size. Must exceed the largest message to be consumed.").
+				Default(10000).
+				Advanced()).
+			Field(service.NewStringEnumField("header_format", "rfc3164", "rfc5424").
+				Description("Specify the syslog message header format.").
+				Default("rfc3164").
+				Advanced()),
+		func(conf *service.ParsedConfig, mgr *service.Resources) (service.Input, error) {
+			return newSyslogUnixDomainSocket(conf, mgr.Logger())
+		})
+	if err != nil {
+		panic(err)
+	}
+}
+
+type syslogUnixReader struct {
+	address       string
+	maxBufferSize int
+	headerFormat  string
+	socketType    string
+	m             sync.RWMutex
+	datagramCn    *net.UnixConn
+	streamingCn   net.Listener
+	buf           []byte
+
+	log *service.Logger
+}
+
+func newSyslogUnixDomainSocket(conf *service.ParsedConfig, log *service.Logger) (p *syslogUnixReader, err error) {
+	p = &syslogUnixReader{
+		log: log,
+	}
+
+	if p.address, err = conf.FieldString("address"); err != nil {
+		return
+	}
+	if p.maxBufferSize, err = conf.FieldInt("max_buffer"); err != nil {
+		return
+	}
+	if p.headerFormat, err = conf.FieldString("header_format"); err != nil {
+		return
+	}
+	if p.socketType, err = conf.FieldString("type"); err != nil {
+		return
+	}
+
+	if p.address == "" {
+		p.log.Errorf("Field address must not be empty")
+		return
+	}
+	switch p.socketType {
+	case "unix":
+		p.streamingCn, err = p.createStreamingSocket()
+		if err != nil {
+			p.log.Errorf("Couldn't create socket: %v", err)
+		}
+	case "unixgram":
+		p.datagramCn, err = p.createDatagramSocket()
+		if err != nil {
+			p.log.Errorf("Couldn't create socket: %v", err)
+		}
+	}
+	return
+}
+
+func (p *syslogUnixReader) createStreamingSocket() (net.Listener, error) {
+	if _, err := os.Stat(p.address); err == nil {
+		if err := os.Remove(p.address); err != nil {
+			p.log.Errorf("Couldn't create socket: %v", err)
+		}
+	}
+	conn, err := net.Listen(p.socketType, p.address)
+	if err != nil {
+		return nil, err
+	}
+
+	c := make(chan os.Signal, 1)
```

Comment:
We can't really allow this within a component implementation since in set ups such as streams mode we need to be able to dynamically spin up and tear down components potentially multiple times. Ideally we want a mechanism for `Close` to trigger this listener clean up logic.